src/components/node-list/index.js

```diff
@@ -105,9 +114,42 @@ const NodeListProvider = ({
     }
   };
 
+  // To determine the parameter name based on the item type
+  const getParamName = (item) => {
+    return isElementType(item.type) ? params.types : params.tags;
+  };
+
+  // To get existing values from search parameters
+  const getExistingValues = (paramName, searchParams) => {
+    const paramValues = searchParams.get(paramName);
+    return new Set(paramValues ? paramValues.split(',') : []);
+  };
+
+  // To update the URL parameters
+  const updateUrlParams = (item, paramName, existingValues) => {
```

Comment:
this logic can go also go in use-generate-pathname.js

Reply:
done.